linux_kernel_modules/rtc_sync/rtc_sync.c

```diff
@@ -0,0 +1,176 @@
+/*
+ * rtc_sync.c: module that syncs sytem time with rtc time 
+ * and vice versa with 10 seconds interval
+ *
+ * Copyright (C) 2018 Sierra Wireless.
+ *
+ * This file is licensed under the terms of the GNU General Public License
+ * version 2.  This program is licensed "as is" without any warranty of any
+ * kind, whether express or implied.
+ */
+
+/* IMPORTANT: the RTC only stores whole seconds. It is arbitrary
+ * whether it stores the most close value or the value with partial
+ * seconds truncated. However, it is important that we use it to store
+ * the truncated value. This is because otherwise it is necessary,
+ * in an rtc sync function, to read both xtime.tv_sec and
+ * xtime.tv_nsec. On some processors (i.e. ARM), an atomic read
+ * of >32bits is not possible. So storing the most close value would
+ * slow down the sync API. So here we have the truncated value and
+ * the best guess is to add 0.5s.
+ */
+
+#include <linux/module.h>
+#include <linux/rtc.h>
+#include <linux/timer.h>
+#include <linux/time.h>
+#include <linux/sched.h>
+#include <linux/kthread.h>
+
+
+static struct task_struct *time_check_thread;
+
+
+//this function syncs system time with RTC when startup - if there is valid time store in RTC
```

Comment:
Linux Kernel coding style requires the use of only C style comments.  eg. `/* a short comment */` and
```
/*
 * This is a longer comment which is too long to fit on a single line without
 * exceeding the 80 character maximum specified in the Linux coding style
 * document (https://www.kernel.org/doc/html/v4.18/process/coding-style.html)
 */
```